### DISCUSSION TITLE
Raw table or custom SQL query ?

### DISCUSSION THREAD

**Discussion by EvanGllt** [Budibase/budibase#14153] [Help]
Hello, there! 👋

My project is a CRUDE application on my database, allowing me to view some data and modify or delete some.

Since the beginning, 95% of my problems are due to the fact that some features are available using (as data source) a raw table and others using a custom SQL query.

I need to make certain data appear that is not present in my database (and I don't want to make it appear in or change its type).

In particular, I have a case where I need a column in my database to be of type "tinyint" but whose display (in my application) is of type "string". So that the field is 0 = "business", 1 = "publiqueAutority". So to solve the problem I perform the following query:

![aide 1](https://github.com/user-attachments/assets/a3c5cb62-58d4-4760-81d3-656462eecb38)

✅ :
![aide 2](https://github.com/user-attachments/assets/08cc8797-7235-487c-9a16-9040c79be1eb)

❌ :
![aide 3](https://github.com/user-attachments/assets/d3420198-8753-4990-9371-3d565d3952ef)

Next to that, I want to be able to see the relationships I've defined:
![aide 4](https://github.com/user-attachments/assets/32670c9b-c2ad-42a8-b532-1e370a6bec0c)

Except I can't find a way of making them appear with a custom query.

So basically, my problem is that I choose one or the other. 😮‍💨

I'm connected to an MS SQL SERVER database, btw.

Could you help me? Thanks in advance.
